storm-core/src/jvm/org/apache/storm/grouping/LoadAwareShuffleGrouping.java

```diff
@@ -36,7 +37,7 @@
 
     @Override
     public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<Integer> targetTasks) {
-        random = new Random();
+        random = new XORShiftRandom();
```

Comment:
Mostly if we are going to use something that is not thread safe here we should explain why it is OK, and what the drawbacks are, because this is supposed to be thread safe.